junit-vintage-engine/src/main/java/org/junit/vintage/engine/discovery/JUnit4DiscoveryRequestResolver.java

```diff
@@ -97,7 +95,7 @@ private TestClassCollector collectTestClasses(EngineDiscoveryRequest discoveryRe
 		};
 	}
 
-	private void populateEngineDescriptor(Set<TestClassRequest> requests) {
+	private void populateEngineDescriptor(Set<TestClassRequest> requests, TestDescriptor engineDescriptor) {
 		new TestClassRequestResolver(engineDescriptor, logger).populateEngineDescriptorFrom(requests);
```

Comment:
`TestClassRequestResolver` is stateful in the same way. You might want to refactor it as well... 😉